landlab/components/flow_accum/lossy_flow_accumulator.py

```diff
@@ -0,0 +1,464 @@
+    cannot accumulate any discharge.
+
+    LossyFlowAccumulator stores as ModelGrid fields:
+
+        -  Node array of drainage areas: *'drainage_area'*
+        -  Node array of discharges: *'surface_water__discharge'*
+        -  Node array of discharge loss in transit (vol/sec). This is the
+            total loss across all of the downstream links:
+            *'surface_water__discharge_loss'*
+        -  Node array containing downstream-to-upstream ordered list of node
+           IDs: *'flow__upstream_node_order'*
+        -  Node array of all but the first element of the delta data structure:
+            *flow__data_structure_delta*. The first element is always zero.
+        -  Link array of the D data structure: *flow__data_structure_D*
+
+    The FlowDirector component will add additional ModelGrid fields; see the
+    FlowAccumulator component for full details. These are:
+
+        -  Node array of receivers (nodes that receive flow), or ITS OWN ID if
+            there is no receiver: *'flow__receiver_node'*
+        -  Node array of flow proportions: *'flow__receiver_proportions'*
+        -  Node array of links carrying flow:  *'flow__link_to_receiver_node'*
+        -  Node array of downhill slopes from each receiver:
+            *'topographic__steepest_slope'*
+        -  Boolean node array of all local lows: *'flow__sink_flag'*
+
+    The primary method of this class is :func:`run_one_step`.
+
+    `run_one_step` takes the optional argument update_flow_director (default is
+    True) that determines if the flow_director is re-run before flow is
+    accumulated.
+
+    Parameters
+    ----------
+    grid : ModelGrid
+        A Landlab grid.
+    surface : field name at node or array of length node
+        The surface to direct flow across.
+    flow_director : string, class, instance of class.
+        A string of method or class name (e.g. 'D8' or 'FlowDirectorD8'), an
+        uninstantiated FlowDirector class, or an instance of a FlowDirector
+        class. This sets the method used to calculate flow directions.
+        Default is 'FlowDirectorSteepest'
+    runoff_rate : field name, array, or float, optional (m/time)
+        If provided, sets the runoff rate and will be assigned to the grid
+        field 'water__unit_flux_in'. If a spatially and and temporally variable
+        runoff rate is desired, pass this field name and update the field
+        through model run time. If both the field and argument are present at
+        the time of initialization, runoff_rate will *overwrite* the field. If
+        neither are set, defaults to spatially constant unit input.
+    depression_finder : string, class, instance of class, optional
+        A string of class name (e.g., 'DepressionFinderAndRouter'), an
+        uninstantiated DepressionFinder class, or an instance of a
+        DepressionFinder class.
+        This sets the method for depression finding.
+    loss_function : Python function, optional
```

Comment:
Two things to be more specific about (here and other places where you list loss_function in the docs).

1) is a positive number a loss or a gain?

2) I think that linkID also means d8ID (if you are on a raster with D8s). We should be more careful with this language. @mcflugen thoughts?

Reply:
1. I think this is clear, due to the way the loss function works: it takes a discharge value, and returns a discharge value. This returned value must always be >0. How the user implements this inside the loss function is their business.
2. Good call.

Reply:
I've also tried to make (1) more clear.